docs/source/deprecated.md

```diff
@@ -1,4 +1,48 @@
-# Deprecated API
+# 0.7 API changes
+
+This page contains information about the major API changes in 0.7.
+
+The previous API reference can be found below.
+
+## Restructuring the library
+
+This is the largest refactoring of the library and there are changes in all parts of the library.
+Other than the three breaking changes below, all changes are backwards compatible, and you will get a deprecation warning with instructions to help porting your code over.
+
+* The library has now been restructured into `iot` and `smart` packages to contain the respective protocol (command set) implementations. The old `Smart{Plug,Bulb,Lightstrip}` that do not require authentication are now accessible through `kasa.iot` package.
+* Exception classes are renamed
+* Exception classes are renamed
+* Using .connect() or discover() is the preferred way to construct device instances rather than initiating constructors on a device.
+
+### Breaking changes
+
+* `features()` now returns a dict of `(identifier, feature)` instead of barely used set of strings.
+* The `supported_modules` attribute is removed from the device class.
+* `state_information` returns information based on features. If you leveraged this property, you may need to adjust your keys.
+
+## Module support for SMART devices
```

Comment:
From here onwards the text reads more like something we should have in the release notes?

Reply:
User friendly text extracted into release notes https://github.com/python-kasa/python-kasa/blob/release/0.7.0/CHANGELOG.md / https://github.com/python-kasa/python-kasa/issues/953